pkg/cloudprovider/vsphere/route/routeprovider_test.go

```diff
@@ -169,6 +169,79 @@ func TestListRoutes(t *testing.T) {
 	assert.Equal(t, "100.96.1.0/24", route.DestinationCIDR, "DestinationCIDR should be 100.96.      1.0/24")
 }
 
+func TestGenerateRoute(t *testing.T) {
+	response := `
+{
+  "results" : [ {
+    "unique_id" : "be13d703-97c9-49f3-818b-91280151b71e",
+    "_last_modified_user" : "admin",
+    "_revision" : 0,
+    "_system_owned" : false,
+    "resource_type" : "StaticRoutes",
+    "_protection" : "NOT_PROTECTED",
+    "_last_modified_time" : 1606960153899,
+    "overridden" : false,
+    "display_name" : "62d347a4-1b70-435e-b92a-9a61453843ee_100.96.0.0_24",
+    "_create_user" : "admin",
+    "network" : "100.96.0.0/24",
+    "tags" : [ {
+      "scope" : "vsphere.k8s.io/cluster-name",
+      "tag" : "kubernetes"
+    }, {
+      "scope" : "vsphere.k8s.io/node-name",
+      "tag" : "node1"
+    } ],
+    "_create_time" : 1606960153895,
+    "path" : "/infra/tier-1s/test-t1/static-routes/62d347a4-1b70-435e-b92a-9a61453843ee_100.96.0.0_24",
+    "marked_for_delete" : false,
+    "enabled_on_secondary" : false,
+    "parent_path" : "/infra/tier-1s/test-t1",
+    "id" : "62d347a4-1b70-435e-b92a-9a61453843ee_100.96.0.0_24",
+    "relative_path" : "62d347a4-1b70-435e-b92a-9a61453843ee_100.96.0.0_24",
+    "next_hops" : [ {
+      "ip_address" : "172.50.0.13",
+      "admin_distance" : 1
+    } ],
+    "status" : {
+      "consolidated_status_per_enforcement_point" : [ {
+        "consolidated_status" : {
+          "consolidated_status" : "SUCCESS"
+        },
+        "resource_type" : "ConsolidatedStatusPerEnforcementPoint",
+        "enforcement_point_id" : "default"
+      } ],
+      "intent_version" : "0",
+      "consolidated_status" : {
+        "consolidated_status" : "SUCCESS"
+      },
+      "intent_path" : "/infra/tier-1s/test-t1/static-routes/62d347a4-1b70-435e-b92a-9a61453843ee_100.96.0.0_24",
+      "publish_status" : "REALIZED"
+    }
+  }],
+  "result_count" : 1,
+  "cursor" : "2"
+}
+`
+	d := json.NewDecoder(strings.NewReader(response))
+	d.UseNumber()
+	var jsondata interface{}
+	d.Decode(&jsondata)
+	decoder := cleanjson.NewJsonToDataValueDecoder()
+	dataValue, _ := decoder.Decode(jsondata)
+	typeConverter := bindings.NewTypeConverter()
+	data, _ := typeConverter.ConvertToGolang(dataValue, bindings.NewReferenceType(model.SearchResponseBindingType))
+
+	p := &routeProvider{}
+	routes := p.generateRoutes(data.(model.SearchResponse))
```

Comment:
Since the other tests are decoding / mocking JSON responses, I feel like this test can be simpler by just passing in a `model.SearchResponse` with only the fields we care about, as opposed to the entire response

Reply:
Can we also add a test case with multiple results from `model.SearchResponse`?

Reply:
Sure, I will remove other unrelated fields.
`TestListRoutes` has 2 results from `model.SearchResponse`

Reply:
I think my first comment wasn't addressed yet but can be follow-up